### PR TITLE
App exit (via ctl app-exit)

### PR DESCRIPTION
**Pull request by gabriel**
Calling `keybase ctl app-exit` will ask the service to tell all app clients to exit.

This will be called by updater before it does a `process.Kill()`.

See https://keybase.atlassian.net/browse/DESKTOP-1552

Also removes the `app.quit()` being triggered from service shutdown.

### CODE REVIEW

**Comment by gabriel** [comment]
cc @keybase/updater-hackers

**Comment by gabriel** [comment]
https://github.com/keybase/go-updater/pull/110

**Comment by cjb** [comment]
Thanks!

> Also removes the app.quit() being triggered from service shutdown.

Why's that?

**Comment by cjb** [comment]
@maxtaco Any idea on a good reviewer for this one? (Maybe you..?)  It's the updater fix we talked about.

File: protocol/avdl/notify_app.avdl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+@namespace("keybase.1")
+protocol NotifyApp {
+
+  @notify("")
+  void exit();
```
Comment (maxtaco) [inline protocol/avdl/notify_app.avdl]: can use `oneway` now!

Comment (maxtaco) [inline protocol/avdl/notify_app.avdl]: I.e.:

```
void exit() oneway;
```

**Comment by maxtaco** [comment]
Small comment, looks solid 👍

**Comment by cjb** [comment]
@gabriel CI failure:
```
../libkb/notify_router.go:419: n.getNotificationChannels(id).App undefined (type keybase1.NotificationChannels has no field or method App)
```

**Comment by gabriel** [comment]
Your question re: app quit from service shutdown is that all services shutdown is triggered by the stop of the app now so that feature is now obsolete.

**Pull request merged**